File: VB/Scaffolding.DetailCollections/DepartmentContextDataModel/DesignTime/DesignTimeDepartmentContextUnitOfWork.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Linq
Imports System.Data
Imports System.Data.Entity
Imports System.Linq.Expressions
Imports System.Collections.Generic
Imports Scaffolding.DetailCollections.Common.Utils
Imports Scaffolding.DetailCollections.Common.DataModel
Imports Scaffolding.DetailCollections.Common.DataModel.EntityFramework
Imports Scaffolding.DetailCollections.Model

Namespace Scaffolding.DetailCollections.DepartmentContextDataModel
	Public Class DepartmentContextDesignTimeUnitOfWork
		Inherits DesignTimeUnitOfWork
		Implements IDepartmentContextUnitOfWork
		Private Shared coursesRepository As New DesignTimeCourseRepository()
		Private Shared departmentsRepository As New DesignTimeDepartmentRepository()
		Private Shared employeesRepository As New DesignTimeEmployeeRepository()

		Public Sub New()
		End Sub

		Private Function HasChanges() As Boolean Implements IDepartmentContextUnitOfWork.HasChanges
			Return False
		End Function
		Private ReadOnly Property Courses() As ICourseRepository Implements IDepartmentContextUnitOfWork.Courses
			Get
				Return coursesRepository
			End Get
		End Property
		Private ReadOnly Property Departments() As IDepartmentRepository Implements IDepartmentContextUnitOfWork.Departments
			Get
				Return departmentsRepository
			End Get
		End Property
		Private ReadOnly Property Employees() As IEmployeeRepository Implements IDepartmentContextUnitOfWork.Employees
			Get
				Return employeesRepository
			End Get
		End Property
		Public Overloads Sub Detach(ByVal entity As Object) Implements Scaffolding.DetailCollections.Common.DataModel.IUnitOfWork.Detach
		End Sub
	End Class
End Namespace

File: CS/Scaffolding.DetailCollections/DepartmentContextDataModel/DesignTime/DesignTimeDepartmentContextUnitOfWork.cs
using System;
using System.Linq;
using System.Data;
using System.Data.Entity;
using System.Linq.Expressions;
using System.Collections.Generic;
using Scaffolding.DetailCollections.Common.Utils;
using Scaffolding.DetailCollections.Common.DataModel;
using Scaffolding.DetailCollections.Common.DataModel.EntityFramework;
using Scaffolding.DetailCollections.Model;

namespace Scaffolding.DetailCollections.DepartmentContextDataModel {
    public class DepartmentContextDesignTimeUnitOfWork : DesignTimeUnitOfWork, IDepartmentContextUnitOfWork {
        static DesignTimeCourseRepository coursesRepository = new DesignTimeCourseRepository();
        static DesignTimeDepartmentRepository departmentsRepository = new DesignTimeDepartmentRepository();
        static DesignTimeEmployeeRepository employeesRepository = new DesignTimeEmployeeRepository();

        public DepartmentContextDesignTimeUnitOfWork() {
        }

        bool IDepartmentContextUnitOfWork.HasChanges() {
            return false;
        }
        ICourseRepository IDepartmentContextUnitOfWork.Courses {
            get { return coursesRepository; }
        }
        IDepartmentRepository IDepartmentContextUnitOfWork.Departments {
            get { return departmentsRepository; }
        }
        IEmployeeRepository IDepartmentContextUnitOfWork.Employees {
            get { return employeesRepository; }
        }
        public void Detach(object entity) {
        }
    }
}
